app/crud.py

```
@@ -0,0 +1,59 @@
+
+
+async def update_todo(db: AsyncSession, todo_id: int, todo: ToDo):
+    try:
+        db_todo = await get_todo(db, todo_id)
+        for key, value in todo.dict(exclude_unset=True).items():
+            setattr(db_todo, key, value)
+        await db.commit()
+        return db_todo
+    except Exception as e:
+        await db.rollback()
+        raise HTTPException(status_code=500, detail=f"Error updating ToDo: {e}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)